stock_split_picking/wizards/stock_picking_split.py

```diff
@@ -0,0 +1,188 @@
+        picking = self.picking_id
+        reserve = False
+        if picking.pack_operation_ids:
+            reserve = True
+            picking.do_unreserve()
+        split_picking = picking.copy(
+            {
+                'name': '/',
+                'move_lines': [],
+                'pack_operation_ids': [],
+            }
+        )
+        # Handle the case where there are several stock.move with the same
+        # product+UoM. We summed the qtys for all these moves in the wizard,
+        # and now we need to spread back the qtys among the different moves
+        # with the same product+UoM.
+        product_split = {}
+        for line in self.line_ids:
+            key = (line.product_id, line.product_uom_id, line.full_split)
+            product_split[key] = line.split_qty
+        new_move_ids = []
+        for move in picking.move_lines:
+            if move.state in ('cancel', 'done'):
+                continue
+            key = (move.product_id, move.product_uom, bool(move.move_orig_ids))
+            to_split = min(product_split[key], move.product_uom_qty)
+            # if to_split > 0
+            if float_compare(to_split, 0, precision_digits=uom_dp) == 1:
+                product_split[key] -= to_split
+                # the 'split()' method expect the qty to be converted in the
+                # UoM of the product
+                to_split_converted = move.product_uom._compute_quantity(
+                    to_split, move.product_id.uom_id)
+                new_move_ids.append(move.split(to_split_converted))
+        self.env['stock.move'].browse(new_move_ids).write(
+            {'picking_id': split_picking.id}
+        )
+        picking.message_post(
+            body=_("Split picking <em>%s</em> <b>created</b>.") %
+            (split_picking.name)
+        )
+        split_picking.message_post(
+            body=_("Split from picking <em>%s</em> <b>created</b>.") %
+            (picking.name)
+        )
+        split_picking.action_confirm()
+        if reserve:
+            picking.action_assign()
+            split_picking.action_assign()
+        return split_picking.id
+
+    def _generate_lines(self):
+        self.ensure_one()
+        line_ids = self.env['stock.picking.split.line']
+        product_qty = {}
+        product_availability = {}
+        for move in self.picking_id.move_lines:
+            if move.state in ('cancel', 'done'):
+                continue
+            reserved_availability = move.product_id.uom_id._compute_quantity(
+                move.reserved_availability, move.product_uom)
+            # key = (Product, UoM, Full split)
+            key = (move.product_id, move.product_uom, bool(move.move_orig_ids))
+            if key not in product_qty:
+                product_qty[key] = move.product_uom_qty
+                product_availability[key] = reserved_availability
+            else:
+                product_qty[key] += move.product_uom_qty
```

Comment:
we can't  do it with products which tracked on serial numbers or lots

Reply:
I updated the PR to add the `stock.move.restrict_lot_id` to the key.